src/main/java/ch/heigvd/dai/models/BMPImage.java

```diff
@@ -26,22 +26,57 @@ public void readImage() {
     try (BufferedInputStream bis = new BufferedInputStream(new FileInputStream(inputPath))) {
       header = new BMPHeader();
       header.readHeader(bis);
+      int width = header.getImageWidth();
+      int height = header.getImageHeight();
+
+      image = new Pixel[width][height];
+
+      for (int h = 0; h < height; h++) { // Iterate from top to bottom
+        for (int w = 0; w < width; w++) {
+          int blue = bis.read();
+          int green = bis.read();
+          int red = bis.read();
+          if (blue == -1 || green == -1 || red == -1) {
+            throw new IOException("Unexpected end of pixel data");
+          }
+          // Store pixels from bottom to top
+          image[w][height - 1 - h] = new Pixel(red, green, blue);
+        }
+      }
 
-      // TODO : read the pixels
     } catch (IOException e) {
       System.err.println("[e]: Error reading image: '" + e.getMessage() + "'");
     } catch (RuntimeException e) {
       System.err.println("[e]: Invalid values: '" + e.getMessage() + "'");
     }
   }
 
-  // TODO : write to a new image
+  public BMPHeader getHeader() {
+    return header;
+  }
+
```

Comment:
Unused +I don't think we need this. We may need it later for image cropping but we'll rewrite it if needed